projects/common/src/time/time-range.service.ts

```diff
@@ -88,46 +87,10 @@ export class TimeRangeService {
     );
   }
 
-  private getPageTimeRangeChanges(): Observable<TimeRange> {
-    return this.navigationService.navigation$.pipe(
-      map(activeRoute => activeRoute.snapshot.queryParamMap),
-      distinctUntilChanged((prevParamMap, currParamMap) => this.shouldNotUpdateTimeRange(prevParamMap, currParamMap)),
-      skip(1), // Skip the first nav, this is handled by initializeTimeRange, but we want to populate prev for distinctUntilChanged
-      filter(queryParamMap => queryParamMap.has(TimeRangeService.TIME_RANGE_QUERY_PARAM)),
-      map(currQueryParamMap => {
-        const timeRangeQueryParamString = currQueryParamMap.get(TimeRangeService.TIME_RANGE_QUERY_PARAM);
-
-        return this.timeRangeFromUrlString(timeRangeQueryParamString!);
-      })
-    );
-  }
-
   private initializeTimeRange(): void {
     this.getInitialTimeRange().subscribe(timeRangeFromInitialUrl => {
       this.applyTimeRangeChange(timeRangeFromInitialUrl);
     });
-
-    this.getPageTimeRangeChanges().subscribe(timeRange => {
```

Comment:
@aaron-steinfeld @arjunlalb I think we need to keep the logic in `getPageTimeRangeChanges()` so that the app can respond to time range changes through URL. We are currently setting new time range from app via this method

```
  private setTimeRangeInUrl(timeRange: TimeRange): this {
    this.navigationService.addQueryParametersToUrl({
      [TimeRangeService.TIME_RANGE_QUERY_PARAM]: timeRange.toUrlString()
    });
    return this;
  }
```

Reply:
Or we modify `this.getInitialTimeRange()` to respond to later tr changes. I am trying to understand why we are using `activeRoute.snapshot` in `getPageTimeRangeChanges`

Reply:
Yeah, probably just poorly named. Can be simplified though now. Can probably just look back in history and revert

Reply:
Good point. No need to reinvent the wheel :)

Reply:
Reverted. `Refresh` query param is also gone.